### DISCUSSION TITLE
When using the dark variant with transition-all, the latter does not work

### DISCUSSION THREAD

**Discussion by NeserCode** [tailwindlabs/tailwindcss#11762] [Help] [answered]
Hello，
As I said, the situation mentioned by title occurs when using the following configuration：

```js
/** @type {import('tailwindcss').Config} */
export default {
  content: [
    "./index.html",
    "./src/**/*.{vue,js,ts,jsx,tsx}"
  ],
  darkMode: 'class',
  theme: {
    extend: {},
  },
  plugins: [],
}
```
I observed from the console of the web page that the previous project used `.dark xxx` to specify the attributes defined in the dark variant, but in the version v3.3.3 I used it was `:is(.dark xxx)` to specify the attributes. Will that make a difference? Or do I need additional configuration?

![image](https://github.com/tailwindlabs/tailwindcss/assets/73767966/162e947c-fac9-47a8-b822-058ebf9fe547)

![image](https://github.com/tailwindlabs/tailwindcss/assets/73767966/f1d434ab-2c2f-4630-a208-b9e03a624706)

**Comment by wongjn** [comment]
The `:is()` wrapping most likely comes from #10835 which was released in `v3.3.0`.

**Reply by NeserCode** [reply]
so, what should i do for solving this problem

**Reply by wongjn** [reply]
What's the problem? Consider providing a [Tailwind Play](https://play.tailwindcss.com) that reproduces the unexpected/undesired behavior.

**Reply by NeserCode** [reply]
emm, links here performs great => https://play.tailwindcss.com/x5iTBPdO6z
However, my application's representation of this problem is that transition-all works fine when using the hover variant, but transition-all doesn't seem to work when using the dark variant.
Then, i check this problem in my application again:

https://github.com/tailwindlabs/tailwindcss/assets/73767966/ad65d400-c628-4ad8-a326-7c98337f3c9d

Could you please give me some advice?

**Reply by NeserCode** [reply]
you maybe need download that video for watching.

**Reply by wongjn** [reply]
Since it works in Tailwind Play, then I would assume that it is some misconfiguration/misuse in your project. Consider providing a remotely-hosted git repository that reproduces the unexpected behavior.

**Reply by NeserCode** [reply]
See [this repo](https://github.com/NeserCode-Studio/Roco-Codex), it's a simple tauri app.
I just finish the titlebar comp and test the dark mode function utils, then it perform wrong, there no more thing except the video shown(i guess). 
:Thinking that the normal tailwind project will like the performance in playground, i provide this repo.
Thanks your patience.

**Reply by wongjn** [reply, marked as answer]
No transitions happen because the `useDark()` toggling functionality from `@vueuse/core` disables transitions by default:

[`useDark()`](https://github.com/vueuse/vueuse/blob/f9f02e562f5f8cb80328670593a6ea66547a87e6/packages/core/useDark/index.ts#L35-L53):
```js
export function useDark(options: UseDarkOptions = {}) {
  // …

  const mode = useColorMode({
    ...options,
    // …
  })
```

Which calls [`useColorMode()`](https://github.com/vueuse/vueuse/blob/f9f02e562f5f8cb80328670593a6ea66547a87e6/packages/core/useColorMode/index.ts#L109-L162):

```ts
export function useColorMode<T extends string = BasicColorMode>(
  options: UseColorModeOptions<T> = {},
): UseColorModeReturn<T> {
  const {
    // …
    disableTransition = true,
  } = options

  // …

  const updateHTMLAttrs = getSSRHandler(
    'updateHTMLAttrs',
    (selector, attribute, value) => {
      // …

      let style: HTMLStyleElement | undefined
      if (disableTransition) {
        style = window!.document.createElement('style')
        const styleString = '*,*::before,*::after{-webkit-transition:none!important;-moz-transition:none!important;-o-transition:none!important;-ms-transition:none!important;transition:none!important}'
        style.appendChild(document.createTextNode(styleString))
        window!.document.head.appendChild(style)
      }
```

This CSS temporarily removes any transitions when switching:

```js
const styleString = '*,*::before,*::after{-webkit-transition:none!important;-moz-transition:none!important;-o-transition:none!important;-ms-transition:none!important;transition:none!important}'
```

So, if you would like to enable transitions when toggling darkmode, you could consider passing `disableTransition: false` when calling `useDark()`, in `src/composables/useDarkMode.ts`:

```diff
 export function useDarkMode() {
 	const isDarkMode = useDark({
 		storageKey: "roco-color-scheme",
+		disableTransition: false,
 	})
 	const toggleDarkMode = useToggle(isDarkMode)
```

**Reply by NeserCode** [reply]
I'm sorry for my omission, and thank you for your reply : )

**Comment by NeserCode** [comment]
Note the problem is:
Using dark variant and `transition`(or `transition-all` etc.) will make the transition effect invalid(i can't see it transform even `duration-1000`).
For example:
```
#app-main {
	@apply w-[500px] h-[600px]
	border border-slate-500
	bg-slate-50 dark:bg-slate-800
	text-slate-700 dark:text-slate-100
	transition-all duration-200 ease-in-out;
}
```
And Using hover variant and `transition`(or `transition-all` etc.) performs great.
Example:
```
.btn {
	@apply inline-flex items-center justify-center w-12 h-full py-2 px-3
	cursor-pointer rounded-none
	text-slate-600 
	bg-slate-100 hover:bg-slate-600
 	transition ease-in-out duration-300;
}
```

**Comment by hxmanh622** [comment]
same problem !

**Reply by wongjn** [reply]
Consider trying the answer given at https://github.com/tailwindlabs/tailwindcss/discussions/11762#discussioncomment-6671551 if you have a similar setup. Otherwise, consider providing a reproducible example.
